### PR TITLE
[Fix] Recreate missing system schema 

### PR DESCRIPTION
**Pull request by Waiwait**
## Changes

Addresses https://github.com/databricks/terraform-provider-databricks/issues/4066.
Makes read function more in line with rest of module to ignore non `enabled` schemas.

## Tests

- [x] `make test` run locally
- [ ] ~relevant change in `docs/` folder~
- [ ] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [ ] ~using Go SDK~

### CODE REVIEW

File: catalog/resource_system_schema.go [hunk 1]
```diff
@@ -93,10 +93,21 @@ func ResourceSystemSchema() common.Resource {
 					if err != nil {
 						return err
 					}
+
+					// only track enabled schemas inline with rest of module
+					if schema.State != catalog.SystemSchemaInfoStateEnableCompleted &&
```
Comment (alexott) [inline catalog/resource_system_schema.go:98]: We also need to handle NotAvailable - these are schemas that you can't enable anymore

Comment (Waiwait) [reply catalog/resource_system_schema.go:98]: What should happen differently in this case? Currently (with this PR), it releases it from state and then assumedly fails when the Create function is called in the next run.

Comment (alexott) [reply catalog/resource_system_schema.go:98]: @Waiwait I think that we need to exclude `UNAVAILABLE` status as well.

Comment (Waiwait) [reply catalog/resource_system_schema.go:97]: Okay, I get it, thanks. Have added.

**Review by alexott — APPROVED** [review]

**Pull request merged**